### PR TITLE
fix(expansion): add expansion panel outputs

### LINKED ISSUE #7172: Md-expansion panel opened and close events not firing on prod build (beta.11)
#### Bug, feature request, or proposal:
Looks like a bug.

#### What is the expected behavior?
Event handler is invoked when opened and closed events happen.

#### What is the current behavior?
Event handler is not invoked on opened and closed events.

#### What are the steps to reproduce?
Attach handlers for such events in the template and check if they are invoked after being built with angular cli's ng build --prod.

#### What is the use-case or motivation for changing an existing behavior?
Has to be working in production build

#### Is there anything else we should know?
I'm attaching event handlers like this:
`<md-expansion-panel
      (closed)="onCollapse()"
      (opened)="onExpand()">`
and process them in component. But these handlers are not invoked after prod build. It works OK when I do ng build --dev or ng serve.

### PR DESCRIPTION
**Pull request by josephperrott**
Fixes #7172

### CODE REVIEW

**Comment by josephperrott** [comment]
@jelbourn  e actually already check this in tests, this error only came to light after `enableProd()` was called on the app.

**Comment by josephperrott** [comment]
@jelbourn This was actually already check for in tests, the error only came to light after `enableProd()` was called in an app using it.

**Comment by danwulff** [comment]
@josephperrott maybe an e2e test would catch this?

**Comment by josephperrott** [comment]
@jelbourn Added this to e2e tests since the tests run using AOT and have AOT tests running in prod mode. Thoughts?

File: src/e2e-app/expansion/expansion-e2e.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<mat-accordion>
```
Comment (jelbourn) [inline src/e2e-app/expansion/expansion-e2e.html]: Instead of adding a separate e2e page here, can you test an example material-examples?

**Comment by josephperrott** [comment]
@jelbourn moved to testing the ExpansionOverviewExample

**Comment by willshowell** [comment]
Friendly ping, since it looks like this is ready for review again

**Comment by jaruesink** [comment]
This seems to be a blocker for anyone wanting to use this component in prod. I was excited when the RC0 came out, because I thought this would be in it.

**Comment by bardiel** [comment]
@jaruesink, in the meantime, you can do something as stated en this comment: https://github.com/angular/material2/issues/7172#issuecomment-332582894

That's just enough to get the expected behavior. Once this gets merged, that workarround will still work, so you'll be able to migrate your code painlessly.

**Comment by hiteshgarg1994** [comment]
<mat-expansion-panel #exeEleRef></mat-expansion-panel>


@ViewChild('exeEleRef') private exeEleRef: any 

  ngAfterViewInit(): void {
    this.exeEleHrly.opened.subscribe(x => {
      // do something
    })
  }

**Pull request merged**